services/auth/source/ldap/source.go

```diff
@@ -52,7 +52,7 @@ type Source struct {
 	GroupFilter           string // Group Name Filter
 	GroupMemberUID        string // Group Attribute containing array of UserUID
 	UserUID               string // User Attribute listed in Group
-	SkipLocalTwoFA        bool   // Skip Local 2fa for users authenticated with this source
+	SkipLocalTwoFA        bool   `json:",omitempty"` // Skip Local 2fa for users authenticated with this source
```

Comment:
I think bool can not be empty ... *bool could do that?

Reply:
If it need be empty, we can use `OptionalBool`, or just like discussed before, use three-value enum to indicate more states. https://github.com/go-gitea/gitea/pull/16954#issuecomment-922325964

Reply:
If this will be exposed as an API, I support still to use `*bool` to make automatically API code generation possible.

Reply:
This is just JSON that is stored in the cfg column.

If it's false just leave it unset and you're not skipping 2fa. It saves parsing time to omit it if it is empty.